lib/factorial.rb

```diff
@@ -1,6 +1,14 @@
 # Computes factorial of the input number and returns it
-# Time complexity: ?
-# Space complexity: ?
+# Time complexity: ? O(n) - linear with n being the number input.
+# Space complexity: ? O(n) with n being the number input for the factorial.
```

Comment:
Since you are not using an array or anything, the space complexity is O(1).  If n gets bigger the amount of memory used doesn't increase.